File: src/tests/test_palindrome.cpp
#include <iostream>
#include <string>

int main() {
    int strLength;
    std::cout << "Please enter the length of the string: ";
    std::cin >> strLength;

    // 使用 std::string 来代替 char 数组
    std::string inputStr;
    std::cout << "Please enter the string: ";
    std::cin >> inputStr;

    // 判断回文
    int midPoint = strLength / 2;
    for (int i = 0; i < midPoint; ++i) {
        // 比较字符串前后对应字符
        if (inputStr[i] != inputStr[strLength - 1 - i]) {
            std::cout << "The string is not a palindrome." << std::endl;
            return 0;
        }
    }

    std::cout << "The string is a palindrome." << std::endl;
    return 0;
}

File: src/tests/test_palindrome.py
from typing import List

def main():
    strLength = None
    print("Please enter the length of the string: ", sep='', end='')
    strLength = int(input())
    inputStr = None
    print("Please enter the string: ", sep='', end='')
    inputStr = str(input())
    midPoint = None
    midPoint = strLength // 2
    for i in range(0, midPoint):
        if inputStr[i] != inputStr[strLength - 1 - i]:
            print("The string is not a palindrome.", "\n", sep='', end='')
            return 0
    print("The string is a palindrome.", "\n", sep='', end='')
    return 0

if __name__ == '__main__':
    main()